src/socket.js

```diff
@@ -0,0 +1,85 @@
+'use strict'
+
+const abortable = require('abortable-iterator')
+const debug = require('debug')
+const log = debug('libp2p:tcp:socket')
+
+const c = require('./constants')
+
+class Libp2pSocket {
```

Comment:
@vasco-santos Isn't this libp2pSocket just a polyfill to add async/await to the Duplex Stream that comes out of utp-native module? If yes, then shouldn't it just be a transform and then attaching the `.getObservedAddrs` method?

Reply:
it was being decided with TCP. We ended up with other approach and I will be updating this PR now